server/github_hook_handler.go

```diff
@@ -74,24 +95,27 @@ func newGithubHookHandler(cc client.GithubClientManager, config *config.Config,
 	}, nil
 }
 
+// Process event
 func (gh *githubHookHandler) processEvent(eventType string, deliveryID string, payload []byte) {
 	eventContext, err := model.ConvertPayloadToEventContext(eventType, payload)
 	if err != nil {
 		log.WithError(err).Error("Error occurred while deserializing request")
 		return
 	}
+
 	if eventContext.GetType() == "tag" {
 		metric.IncreaseCounter(metric.TagRequestCount)
 	} else {
 		metric.IncreaseCounter(metric.BranchRequestCount)
 	}
+
 	eventContext.Log()
 
-	if "workflow_run" == eventContext.GetEvent() && "completed" == eventContext.GetAction() {
-		if err := gh.ClientManager.RevokeToken(eventContext.GetRepository(), eventContext.GetWorkflowRunID()); err != nil {
-			log.WithError(err).Error("Error occurred while revoking pipeline token")
-		}
-	}
+	// if "workflow_run" == eventContext.GetEvent() && "completed" == eventContext.GetAction() {
```

Comment:
I am not so sure . I think we will just complicate things . Since this tool is using cache and a queueing system we might have a scenario where the token will be revoked while the bot is polling for the workflow status . 

I can leave this with a comment and a hyperlink to the logic.